.github/workflows/ci.yaml

```diff
@@ -27,7 +27,7 @@ jobs:
           version: v3.6.3
       - uses: actions/setup-python@v4
         with:
-          python-version: 3.7
+          python-version: 3.12
```

Comment:
this is because github changed the ubuntu OS so now only python 3.9-3.12 are supported